kymatio/scattering1d/scattering1d.py

```diff
@@ -355,11 +355,17 @@ def forward(self, x):
             is `False`, it is a dictionary indexed by tuples of filter indices.
         """
         # basic checking, should be improved
-        if len(x.shape) != 2:
+        if x.dim() < 1:
             raise ValueError(
-                'Input tensor x should have 2 axis, got {}'.format(
-                    len(x.shape)))
-        x = x.unsqueeze(1)
+                'Input tensor x should have at least one axis, got {}'.format(
+                    x.dim()))
+
+        batch_shape = x.shape[:-1]
+        signal_shape = x.shape[-1:]
+
+        batch_length = int(np.prod(batch_shape))
+
+        x = x.view((batch_length, 1) + signal_shape)
```

Comment:
`reshape` should probably be fine and is more framework-agnostic.

Also I'd remove `batch_length` in favor of `-1` if it isn't used anywhere else